### ISSUE TITLE
Escape Character Does Not Get Escaped

### ISSUE THREAD

**Issue by waylonflinn** [wdavidw/node-csv-stringify#24]
When using the `escape` option, if the provided escape character is present in the input, it is not escaped.

Here's an example.

``` javascript
var stream = require('stream'),
    stringify = require('csv-stringify'),
    parse = require('csv-parse');

var stringifier = stringify({"header":true, "quoted":true, "escape":"\\"});

var parser = parse({"columns": true, "escape":'\\'});

var input = { "one": "test", "two": 'This a string with \\"backslashes\\" in it'};

var rs = new stream.Readable({ objectMode: true });

rs.push(input);
rs.push(null);
rs.pipe(stringifier)
    .pipe(parser);
```

This is a little hard to follow but this string, `'This a string with \\"backslashes\\" in it'`, has a single backslash (escaped in javascript) followed by a quote, which occurs two times. 

When you `console.log` it, it looks like this:
`'This a string with \"backslashes\" in it'`
## Expected

What should happen is that this string gets turned into a string like this:
`'This a string with \\\"backslashes\\\" in it'`

That's one escape character for the already present `\` and a second for the `"`.
## Observed

What actually happens is that it gets turned into a string like this:
`'This a string with \\"backslashes\\" in it'`

one escape character for the delimiter `"` and nothing for the already existing escape character.

**Comment by DullReferenceException** [comment]
Any updates on this? This looks like a pretty important fix.

**Comment by wdavidw** [comment]
Is the problem with the parser or the stringifier? Looking at it, i'm not sure we shall escape double if inside a single quote string, no?

**Comment by DullReferenceException** [comment]
Looks like the stringifier; in our code, where we use these options:

```javascript
{
  header: true,
  escape: '\\',
  delimiter: ',',
  columns: [...]
}
```

...we're getting output where a `\` in the string is not getting escaped.

**Comment by DullReferenceException** [comment]
I thought I identified this problem as well, but I added a unit test to prove it, and the test passed. It turned out in our case that we were not using the same escape options on parse as on stringifying. Not sure if @waylonflinn had a similar issue or not.

**Comment by wdavidw** [comment]
is the unit test you wrote worth adding to this code base? if so, i can include it.

**Comment by DullReferenceException** [comment]
Yep, will do.

**Comment by wdavidw** [comment]
doing some house cleaning, closing.
